src/env-inl.h

```diff
@@ -250,6 +280,22 @@ inline uv_check_t* Environment::idle_check_handle() {
   return &idle_check_handle_;
 }
 
+inline void Environment::DeregisterHandleCleanup(uv_handle_t* handle) {
+  HandleCleanup* to_remove = nullptr;
+  // Calling Remove() while in-loop will make it infinite.
+  for (HandleCleanup* hc : handle_cleanup_queue_) {
+    if (hc->handle_ == handle) {
+      to_remove = hc;
+      break;
+    }
+  }
```

Comment:
Yes I thought about that and we should consider changing this to a structure that allows faster removal